.github/scripts/trymerge.py

```diff
@@ -1265,6 +1295,90 @@ def checks_to_str(checks: List[Tuple[str, Optional[str]]]) -> str:
 def checks_to_markdown_bullets(checks: List[Tuple[str, Optional[str]]]) -> List[str]:
     return [f"- [{c[0]}]({c[1]})" if c[1] is not None else f"- {c[0]}" for c in checks[:5]]
 
+def get_flaky_rules() -> List[FlakyRule]:
+    url = "https://raw.githubusercontent.com/pytorch/test-infra/generated-stats/stats/flaky-rules.json"
+    flaky_rules_json: List[FlakyRule] = []
+    for _ in range(3):
+        try:
+            raw_json = fetch_json_list(url)
+            for rule in raw_json:
+                flaky_rules_json.append(FlakyRule(**rule))
+            break
+        except Exception as e:
+            print(f"Could not download {url} because: {e}.")
+    return flaky_rules_json
+
+def get_rockset_results(head_sha: str, merge_base: str) -> List[Dict[str, Any]]:
+    query = f"""
+SELECT
+    w.name as workflow_name,
+    j.id,
+    j.name,
+    j.conclusion,
+    j.completed_at,
+    j.html_url,
+    j.head_sha,
+    j.torchci_classification.captures as failure_captures,
+    LENGTH(j.steps) as steps,
+FROM
+    commons.workflow_job j join commons.workflow_run w on w.id = j.run_id
+where
+    j.head_sha in ('{head_sha}','{merge_base}')
+"""
+    res = rockset.Client(
+        api_server="api.rs2.usw2.rockset.com", api_key=os.environ["ROCKSET_API_KEY"]
+    ).sql(rockset.Q(query))
+    return cast(List[Dict[str, Any]], res.results())
+
+def get_classifications(
+    head_sha: str,
+    merge_base: str,
+    checks: Dict[str, JobCheckState]
+) -> Dict[str, JobCheckState]:
+    flaky_rules_json = get_flaky_rules()
+
+    rockset_results = get_rockset_results(head_sha, merge_base)
+    head_sha_jobs: Dict[str, Dict[str, Any]] = {}
+    merge_base_jobs: Dict[str, Dict[str, Any]] = {}
+
+    def insert(d: Dict[str, Dict[str, Any]], key: str, val: Dict[str, Any]) -> None:
+        if key not in d:
+            d[key] = val
+            return
+        if d[key]["conclusion"] == "success":
```

Comment:
Would there be an edge case where people re-run the whole workflow manually and get a flaky failure on a job that was success originally?  Do you think it's simpler just to store the conclusion from the latest run regardless of its success or failure status?